### ISSUE TITLE
FEAT: Support to enforce conventional commits, custom scopes and amend other tool commits

### ISSUE THREAD

**Issue by D4RKAR117** [di-sukharev/opencommit#61]
Hey, first of all awesome tool

I want to know if there is something about conventional commits (or any other preset like angular) into the tool, would be awesome for workflows that use extensively commit lint.

Also, there is some way that you can choose the scope? Would be great for monorepos where scopes usually belongs to the package that is being modified.

Finally, would be great if this tool can amend the commit after the manual writing of the commit, this could help for integration with other git tools like commitizen.

Keep the awesome work!!

**Comment by di-sukharev** [comment]
you can do `oc --amend` and all other git commit flags are also passed to `oc`. Sorry, I dont get your first 2 points :0

**Comment by D4RKAR117** [comment]
Yeah, by default some tools like monorepos use git scopes to automate releases of certain packages, so would be great that as @jakob1379 suggest, the AI uses conventional commits or any other git spec (like angular commits)  or even custom specs to do their job, so tools that sues that spec don't fail, something like commitlint and that

**Comment by di-sukharev** [comment]
so you are saying that adding a blank space between header and description would solve the issue?

**Comment by dasheck0** [comment]
+1 this. I maintain multiple projects with their respective commit message style. It is not always conventional commits sometimes just plain english (what I dislike. But this is a different discussion altogether). However, it would be great if OC recognizes this and creates messages according to the commit style that is already present. As far as I know one can send this is further context for GPT-3.5. I suppose that last `n` commits are sufficient as well, so that you don't have to send the while history for this to work.

**Comment by di-sukharev** [comment]
could anybody help me here please and make the PR? <3

**Comment by dasheck0** [comment]
I can take a look it this. Not sure when I get to it. Probably this weekend

**Comment by sebastienfi** [comment]
> +1 this. I maintain multiple projects with their respective commit message style. It is not always conventional commits sometimes just plain english (what I dislike. But this is a different discussion altogether). However, it would be great if OC recognizes this and creates messages according to the commit style that is already present. As far as I know one can send this is further context for GPT-3.5. I suppose that last `n` commits are sufficient as well, so that you don't have to send the while history for this to work.

I'd rather have a config to define the style, because previous commit messages might be flawed.

**Comment by sebastienfi** [comment]
I'm working on this, I will submit a PR soon.

**Comment by sebastienfi** [comment]
@D4RKAR117 @jakob1379 @dasheck0  Could you please review the PR and provide feedback?

**Comment by di-sukharev** [comment]
@sebastienfi hi, are we done here or should i reopen?

**Comment by di-sukharev** [comment]
ok, i see <3

**Comment by Mikaleb** [comment]
Anything on this ?

**Comment by eyyyyyyy3** [comment]
if you want to use AI to create you commits, there is [cz-git](https://github.com/Zhengqbbb/cz-git). It also has ollama support. Still it would be nice if there were a commitizen opencommit adapter but I guess that conversation would be more appropriate at the commitizen issues page.
